src/parser.rs

```diff
@@ -216,11 +263,56 @@ impl<'i> Iterator for Input<'i> {
     }
 }
 
+/// Wrapper for syntax violation callback functions.
+#[derive(Copy, Clone)]
+pub enum ViolationFn<'a> {
```

Comment:
Make this `pub(crate)`.

Reply:
Changing this to `pub(crate)` fails with the following errors on the parser module public, but crate private, structs `Input` and `Parser`.  I suspect that all of these cases would need to be made `pub(crate)` of just `pub`, but I'm reluctant to add that this PR, as it expands scope and risk. If it worth it, consider as a follow-on change?

```txt
error[E0446]: private type `parser::ViolationFn<'_>` in public interface
   --> src/parser.rs:166:5
    |
166 | /     pub fn with_log(original_input: &'i str, vfn: ViolationFn) -> Self {
167 | |         let input = original_input.trim_matches(c0_control_or_space);
168 | |         if vfn.is_set() {
169 | |             if input.len() < original_input.len() {
...   |
176 | |         Input { chars: input.chars() }
177 | |     }
    | |_____^ can't leak private type

error[E0446]: private type `parser::ViolationFn<'a>` in public interface
   --> src/parser.rs:319:5
    |
319 |     pub violation_fn: ViolationFn<'a>,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type
~~~

Reply:
I guess that's ok, I just wanted to make sure `ViolationFn` doesn't escape the crate.